unmarshal.go

```diff
@@ -36,7 +36,7 @@ func unmarshalWithPrefix(data interface{}, prefix string) error {
 			continue
 		}
 
-		if err := unmarshalField(field, tag, prefix); err != nil {
+		if err := unmarshalField(field, tag, prefix, data); err != nil {
```

Comment:
It's getting too complicated, it was OK when you had 2 fields

you should define a struct with all the fields you need, then create a variable from this struct, then pass a reference to the variable and, that's it

No more code rewrites when adding an new feature, and also less pointers

Reply:
There wasn't really a way to make the default value lookup work without this. I'm open to suggestions, but variable expansion is a strong feature of environment variables.

Reply:
I could make a suggestion, but I have to get back from holidays, as I only have a phone with me right now.

So, I'm opening a refactoring issue

Reply:
I created 

- #10